packages/npm-packages/ruby-wasm-wasi/test/unit/test_object.rb

```diff
@@ -193,6 +193,14 @@ def test_new_standard_built_in_object_with_js_string
     assert_equal "hello", JS.global[:String].new(JS.eval("return 'hello'")).to_s
   end
 
+  def test_new_standard_built_in_object_with_block
+    assert_equal JS::Undefined.to_s,
```

Comment:
Looks good 👍 Could you cover usual args + block case? Also the use of Promise and setTimeout is great as an integration test, but I'd like to have simpler and more generic cases without use of specific APIs (but the Promise test is worth leaving as is at the tail of this test case)